docs/release-notes/release-notes-0.9.6.md

```diff
@@ -0,0 +1,192 @@
+Joinmarket-clientserver 0.9.6:
+=================
+
+<https://github.com/joinmarket-org/joinmarket-clientserver/releases/tag/v0.9.6>
+
+Please report bugs using the issue tracker at github:
+
+<https://github.com/joinmarket-org/joinmarket-clientserver/issues>
+
+
+Upgrading
+=========
+
+To upgrade:
+
+*Reminder: always back up and recreate your joinmarket.cfg file when doing the upgrade; this is to make sure you have the new default settings.*
+
+(If you are upgrading from a version pre-0.7.0 please read the "Upgrading" section in [the 0.7.0 release notes](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/release-notes/release-notes-0.7.0.md).)
+
+(If you are upgrading from a version pre-0.9.0 please read the [release notes](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/release-notes/release-notes-0.9.0.md) to find out how about [fidelity bonds](../fidelity-bonds.md) are relevant to your use-case).
+
+First run the `install.sh` script as mentioned in the README. When prompted to overwrite the directory `jmvenv`, accept.
+
+Note that `./install.sh -?` will show the options for installation. See the section on Installation below for a new option for Tor.
+
+Notable changes
+===============
+
+### New p2p onion messaging channels, with directory nodes
+
+**Do not forget to re-create your joinmarket.cfg to activate this!**
+
+This is an implementation of the ideas first laid out in [#415](https://github.com/JoinMarket-Org/joinmarket-clientserver/issues/415), i.e. less or non-reliance on central servers for passing messages. That reliance has disadvantages even considering that the transaction negotiation is end to end encrypted, in regards to speed, scalability, censorship resistance and metadata leakage. So while IRC servers, used redundantly, as pre-0.9.6 Joinmarket, works, it has limitations.
+**This does not (yet) mean we stopped using IRC** - this is an added message channel, and all message channels are used redundantly. The user can choose to use any set of message channels they prefer, but the defaults are obviously the best choice for most.
+From this release onwards, we have multiple (currently three, managed by @AdamISZ @bisqubutor @openoms) "directory node" onion addresses which serve a similar function to IRC servers with a difference: once takers find makers' onion addresses via this server, they communicate peer to peer. This should result in faster transaction negotiation and the possibility of much larger anon-set coinjoins getting negotiated, as well as being a little better as a privacy model.
```

Comment:
That's an interesting perspective, I think I don't agree - having these run by trusted members of the community just helps a little bit to 'bind' the network effect of them, while on the other side, they don't hold coins or blockchain access or control coinjoins at all. The only risk (and even then it's slight, due to the substantial redundancy) is that they might censor. For the dnode runners themselves, there isn't really much security risk as far as I can see, since it's onion only.

Fwiw I already send out my identity in the handshake, and I was even thinking to gpg sign the onion address itself (haven't got round to it yet).

Reply:
OK in the absence of other comments I'm going to go with what we have.

It's easy for us to use or switch to anonymous d-nodes instead.

It's also easy for people to randomly set up their own d-nodes without telling us. For now I think this is the right decision.

Reply:
I'm ok with this for now, not such a big deal.

In future would be cool to have hardcoded list only for initial bootstrap, then bots could somehow send directory node info to each other that could be cached locally.

Reply:
Maybe. Thing is, this is very different from e.g. bitcoin bootstrap. Because people want access to all the other bots, whereas in bitcoin it's a very flat p2p structure and as long as you have a *decent* number of heterogeneous peers it should be good enough (I mean, I know there's more to it than that with isolation attacks etc.). But in Joinmarket it's really important for a bot to know it's where other bots are (both taker and maker side, but especially maker side). So I think there's a more binding network effect here.